### PR TITLE
Added screenshot functionality

### PR DESCRIPTION
**Pull request by dpgraham**

### CODE REVIEW

File: lib/simctl.js [hunk 1]
```diff
@@ -236,5 +236,13 @@ async function getDevices (forSdk:string = null):Object {
   return devices;
 }
 
+async function getScreenshot (udid:string):string {
+  let tempScreenshotPath = 'temp-screenshot-' + udid + '.png';
```
Comment (imurchie) [inline lib/simctl.js]: Where is this file created? Should it be in a temporary directory?

Comment (dpgraham) [inline lib/simctl.js]: It's created in whatever the `cwd` is. Should it be created elsewhere?

Comment (imurchie) [inline lib/simctl.js]: I think it would be better to put it in a temp directory.

https://github.com/appium/appium-support/blob/master/lib/tempdir.js

Comment (dpgraham) [inline lib/simctl.js]: I might actually be able to send it to stdout

```
	screenshot [--type=<type>] [--display=<display>] <file or url>
	    Saves a screenshot as a PNG to the specified file or url(use "-" for stdout).
	    --type       Can be "png", "tiff", "bmp", "gif", "jpeg". Default is png.

	    --display    iOS: supports "main" or "tv". Default is "main".
	                 tvOS: supports only "tv"
	                 watchOS: supports only "main"

```

**Comment by dpgraham** [comment]
@imurchie Yeah I can make a comment on that. The xcuitest-driver just uses a try-catch and then falls back to the WDA proxy when it isn't supported.

**Comment by dpgraham** [comment]
Changes made

File: lib/simctl.js [hunk 2]
```diff
@@ -236,5 +237,19 @@ async function getDevices (forSdk:string = null):Object {
   return devices;
 }
 
+/**
+ * Gets base64 screenshot for device (xcode >= 8.1 only)
+ * @param {string} udid 
+ */
+async function getScreenshot (udid:string):string {
+  let tempScreenshotDir = path.resolve(__dirname, 'tmp-screenshots');
+  await fs.mkdir(tempScreenshotDir);
```
Comment (imurchie) [inline lib/simctl.js]: This stuff is all done correctly using https://github.com/appium/appium-support/blob/master/lib/tempdir.js

**Comment by dpgraham** [comment]
I added the 'tempDir' library. I still rimraf the screenshot png just so that it's not sitting on the user's filesystem.

**Comment by imurchie** [comment]
That's probably a good idea. But this way whatever the case it isn't randomly in the Appium installation.

**Comment by dpgraham** [comment]
Yeah, the tempdir solution is much better. I

**Comment by dpgraham** [comment]
Change published to 3.5.5

**Pull request merged**